### PR TITLE
components: Use vanilla emotion for `view`

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Use vanilla emotion for the base `View` component... as well as a whole slew of other updates that are necessary to makae this initial PR to migrate to vanilla emotion work.

I'm sorry for the size of this PR. I tried to split it up but it seems impossible, there are either various build failures or tests that won't pass without all of these changes. I wish I could summarize the changes somehow but honestly it followed the process of starting with the smallest change and then following every rabbit hole that opened up until no more rabbit holes existed. I've essentially played whack-a-mole and whacked all the moles 😭 

I'd be happy to walk anyone though this PR and the progression of it. Hopefully the commits help a little bit with that, but some of them aren't split up nicely at all.

Even with all these changes, there's still a lot to do. This leaves certain components like `surface` completely broken, but it's also unused so I think this is safe to do.

## How has this been tested?
Run `npm run storybook:dev` and ensure that G2 components continue to work, especially thinks like Text and Flex which use `createComponent` as these depend directly on `View`.

Also the tests and build should pass.

## Types of changes
New feature?

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by sarayourfriend** [comment]
Note, this PR is currently in a broken state, i'm not sure how to fix it as of right now but am brainstorming ideas. The core of the problem is that by not importing from `@wp-g2/styles` the variables are now all undefined, I think we just need to add them back into the mixin that adds CSS variables in SASS but I'm not sure yet.

File: packages/components/src/ui/card/component.js [hunk 1]
```diff
@@ -12,20 +11,21 @@ import { useMemo } from '@wordpress/element';
...
- * @param {import('@wp-g2/create-styles').ViewOwnProps<import('./types').CardProps, 'div'>} props
+ * @param {import('../context').ViewOwnProps<import('./types').CardProps, 'div'>} props
  * @param {import('react').Ref<any>} forwardedRef
  */
 function Card( props, forwardedRef ) {
 	const { children, elevation, isRounded = true, ...otherProps } = useCard(
 		props
 	);
-	const elevationBorderRadius = isRounded ? ui.get( 'cardBorderRadius' ) : 0;
+	const elevationBorderRadius = isRounded ? '2px' : 0;
```
Comment (youknowriad) [inline packages/components/src/ui/card/component.js]: don't we have a JS variable for the default border radius or something like that.

**Review by youknowriad — APPROVED** [review]
I love all the removed code, keep it coming :P

File: packages/components/src/ui/card/component.js (hunk 1 above)
Comment (sarayourfriend) [reply packages/components/src/ui/card/component.js]: Yup, I'm not sure why I didn't use it here 🤦‍♀️

**Pull request merged**